lib/python/picongpu/plugins/plot_mpl/phase_space_visualizer.py

```diff
@@ -0,0 +1,158 @@
+            path to the run directory of PIConGPU
+            (the path before ``simOutput/``)
+        """
+        if run_directory is None:
+            raise ValueError('The run_directory parameter can not be None!')
+
+        # initialize the data-reader
+        self.phase_space = PhaseSpace(run_directory)
+        self.plt_obj = None
+
+        # for unit-conversion from SI (taken from picongpu readthedocs)
+        self.mu = 1.e6
+        self.e_mc_r = 1. / (9.1e-31 * 2.9979e8)
+
+    def visualize(self, iteration, ax, species='e', species_filter='all',
+                  ps='ypy', **kwargs):
+        """
+        Creates a phase space plot on the provided axes object for
+        the data of the given iteration using matpotlib.
+
+        Parameters
+        ----------
+        iteration: int
+            the iteration number for which data will be plotted.
+        ax: matplotlib axes object
+            the part of the figure where this plot will be shown.
+        species : string
+            short name of the particle species, e.g. 'e' for electrons
+            (defined in ``speciesDefinition.param``)
+        species_filter: string
+            name of the particle species filter, default is 'all'
+            (defined in ``particleFilters.param``)
+        ps : string
+            phase space selection in order: spatial, momentum component,
+            e.g. 'ypy' or 'ypx'
+        kwargs: dict
+            possible additional keyword args (e.g. for styling).
+            NOTE: no options from this parameter are considered yet!
+        """
+        data, meta = self.phase_space.get(
+            species=species,
+            species_filter=species_filter,
+            ps=ps,
+            iteration=iteration)
+
+        if self.plt_obj is None:
+            self.plt_obj = ax.imshow(
+                np.abs(data).T * meta.dV,
+                extent=meta.extent *
+                [self.mu, self.mu, self.e_mc_r, self.e_mc_r],
+                interpolation='nearest',
+                aspect='auto',
+                origin='lower',
+                norm=LogNorm()
+            )
+        else:
+            self.plt_obj.set_data(np.abs(data).T * meta.dV)
+
+        # prevent multiple rendering of colorbar
+        if not self.plt_obj.colorbar:
+            cbar = plt.colorbar(self.plt_obj, ax=ax)
+            cbar.set_label(
+                r'$Q / \mathrm{d}r \mathrm{d}p$ [$\mathrm{C s kg^{-1} m^{-2}}'
+                '$')
+
+        ax.set_xlabel(r'${0}$ [${1}$]'.format(meta.r, "\mathrm{\mu m}"))
+        ax.set_ylabel(r'$p_{0}$ [$\beta\gamma$]'.format(meta.p))
+
+
+if __name__ == '__main__':
+
+    def main():
+
+        import sys
+        import getopt
+
+        def usage():
+            print("usage:")
+            print(
+                "python", sys.argv[0],
+                "-p <path to run_directory> -i <iteration>"
+                " -s <particle species> -f <species_filter>"
+                " -m <momentum term>")
```

Comment:
`momentum term` is actually both the `spatial component` and the `momentum component`.

`-m` is ok with me, but add `-m <phase space selection>` as in the data api